pkgs/tools/misc/hdf4/default.nix

```diff
@@ -31,11 +31,12 @@ stdenv.mkDerivation rec {
     "-DHDF4_BUILD_WITH_INSTALL_NAME=OFF"
     "-DHDF4_ENABLE_JPEG_LIB_SUPPORT=ON"
     "-DHDF4_ENABLE_NETCDF=OFF"
-    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
-    "-DHDF4_ENABLE_SZIP_SUPPORT=ON"
     "-DHDF4_ENABLE_Z_LIB_SUPPORT=ON"
     "-DHDF4_BUILD_FORTRAN=OFF"
     "-DJPEG_DIR=${libjpeg}"
+  ] ++ stdenv.lib.optionals (szip != null) [
+    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
```

Comment:
I do not understand. In this nix package, we are always building from source!? Do you mean this package should behave differently on Hydra than when building locally? Is that even possible? Even if so, this would be very confusing.

Even if we are building from source, I expect that Nixpkgs does not contain any unfree software by default. As PostGIS is free software and only some small optional functionality depends on unfree software, I expect PostGIS to only contain free software by default.

Also, the license does not distinguish between using a prebuilt binary or building szip yourself, so I don't see how that makes any difference?